File: backend/src/main/java/com/xyehyin/hexuanning/entity/Category.java
package com.xyehyin.hexuanning.entity;

import jakarta.persistence.*;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import org.springframework.data.annotation.CreatedBy;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;
import java.util.List;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "category")
@EntityListeners(AuditingEntityListener.class)
public class Category {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 50)
    private String name; // 分类名称

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parent_id")
    private Category parent; // 父级分类（可选）

    @OneToMany(mappedBy = "parent", cascade = CascadeType.REMOVE)
    private List<Category> children; // 子分类（级联）

    @Column(length = 255)
    private String description; // 描述（可选）

    @Column(nullable = false)
    private Boolean enabled = true; // 是否启用

    @CreatedDate
    @Column(updatable = false)
    private LocalDateTime createTime;

    @LastModifiedDate
    private LocalDateTime updateTime;
}

File: frontend/src/api/types/category.ts
import type { ApiResponse } from './common'

export interface CreateCategoryDTO {
  name: string
  parentId?: number
  description?: string
}

export interface UpdateCategoryDTO {
  name?: string
  parentId?: number
  description?: string
}

export interface CategoryFlatVO {
  id: number
  name: string
  description: string
  parent?: CategoryFlatVO
  enabled: boolean
  createTime: string
  updateTime: string
}

export interface CategoryTreeVO {
  id: number
  name: string
  description: string
  enabled: boolean
  createTime: string
  updateTime: string
}

export type ApiResponseCategoryFlatVO = ApiResponse<CategoryFlatVO>
export type ApiResponseListCategoryFlatVO = ApiResponse<CategoryFlatVO[]>
export type ApiResponseListCategoryTreeVO = ApiResponse<CategoryTreeVO[]>
